Docs/Architecture.md

```diff
@@ -0,0 +1,1429 @@
+
+### MU_BASECORE
+
+The MU_BASECORE repository contains the core UEFI/EDK2 code with OneCrypto integration points. Changes here define the public interfaces that platforms and modules use.
+
+**Key Components:**
+
+- **`CryptoPkg/Include/Library/BaseCryptLib.h`**
+  - Defines the standard cryptographic API interface used by all UEFI modules
+  - This is the stable API that modules depend on
+  - No knowledge of OneCrypto implementation details
+
+- **`CryptoPkg/Include/Protocol/OneCrypto.h`**
+  - Defines the `ONE_CRYPTO_PROTOCOL` structure with versioned crypto function pointers
+  - Contains protocol GUIDs: `gOneCryptoDxeProtocolGuid`, `gOneCryptoMmProtocolGuid`, `gOneCryptoPrivateProtocolGuid`
+  - Version fields: `UINT16 Major`, `UINT16 Minor`
+  - All crypto function signatures and documentation
+
+- **`CryptoPkg/Library/BaseCryptLibOnOneCrypto/`**
+  - Implementation of BaseCryptLib that locates and calls OneCrypto protocol
+  - Phase-specific instances:
+    - `BaseCryptLibOnOneCrypto_Dxe` - Locates DXE protocol via Boot Services
+    - `BaseCryptLibOnOneCrypto_Mm` - Locates MM protocol via MM Services
+  - Checks protocol version before calling functions to ensure compatibility
+  - Thin wrapper layer with minimal logic
+
+- **`CryptoPkg/Library/BaseCryptLibNull/`**
+  - Null implementation that returns errors for all functions
+  - Used when crypto is not available or during early boot phases
+
+### CryptoBinPkg
+
+CryptoBinPkg builds the small loader drivers that find, load, and initialize `OneCryptoMmBin.efi` in each phase. These loaders have no build-time dependency on OpenSSL or OneCryptoPkg.
+
+**Key Components:**
+
+- **`Driver/OneCryptoLoaderDxe.c` / `OneCryptoLoaderDxe.inf`**
+  - DXE phase loader driver (~15 KB)
+  - Searches the FV for OneCryptBin
+  - Loads the binary into memory that DXE can access
+  - Provides phase specific dependencies to OneCryptoBin
+  - Publishes `ONE_CRYPTO_PROTOCOL`
+
+- **`Driver/OneCryptoLoaderMm.c` / `OneCryptoLoaderMm.inf`**
+  - MM phase loader driver (~4 KB)
+  - Locates the private protocol shared by the OneCryptoBin entry
+  - Provides phase specific dependencies to OneCryptoBin
+  - Publishes `ONE_CRYPTO_PROTOCOL`
+
+- **`Support/OneCryptoDxeLoader.inf`** and **`Support/OneCryptoLoaderMm.inf`**
+  - Library-style INF files for platform integration
+  - Allow platforms to include loaders as dependencies
+
+**Build Outputs:**
+- `OneCryptoDxeLoader.efi` - DXE loader (~15 KB uncompressed)
+- `OneCryptoLoaderMm.efi` - MM loader (~4 KB uncompressed)
+- `OneCryptoBinMm.efi` - Phase agnostic crypto provider (~1.4 MB uncompressed)
+
+### OneCryptoPkg
+
+OneCryptoPkg builds the standalone crypto binary (`OneCryptoMmBin.efi`) that contains all cryptographic functionality. This binary is phase-independent and loaded by the loaders.
+
+**Key Components:**
+
+- **`OneCryptoMmBin/`**
+  - Main binary application that provides crypto services
+  - Links against OpensslPkg's BaseCryptLib and OpensslLib
```

Comment:
To clarify: This should just be `BaseCryptLib`. How that instance is satisfied (whether with OpenSSL or something else) is beyond the scope of this module's details.